spec/namespaces/flow_framework.yaml

```diff
@@ -0,0 +1,261 @@
+      parameters:
+        - $ref: '#/components/parameters/flow_framework.delete::query.clear_status'
+      responses:
+        '200':
+          $ref: '#/components/responses/flow_framework.delete'
+        '403':
+          description: if workflow id is null.
+    put:
+      operationId: flow_framework.update.0
+      x-operation-group: flow_framework.update
+      description: Update a workflow. You can only update a complete workflow if it has not yet been provisioned.
+      externalDocs:
+        url: https://opensearch.org/docs/latest/automating-configurations/api/create-workflow/
+      parameters:
+        - $ref: '#/components/parameters/flow_framework.update::query.update_fields'
+        - $ref: '#/components/parameters/flow_framework.get::path.workflow_id'
+      requestBody:
+        $ref: '#/components/requestBodies/flow_framework.update'
+      responses:
+        '201':
+          $ref: '#/components/responses/flow_framework.create'
+          description: It returns the workflow_id
+    get:
+      operationId: flow_framework.get.0
+      x-operation-group: flow_framework.get
+      description: Get a workflow.
+      parameters:
+        - $ref: '#/components/parameters/flow_framework.get::path.workflow_id'
+      responses:
+        '200':
+          $ref: '#/components/responses/flow_framework.get'
+
+
+components:
+  parameters:
+    flow_framework.get::path.workflow_id:
+      name: workflow_id
+      in: path
+      required: true
+      schema:
+        type: string
+    flow_framework.create::query.provision:
+      in: query
+      name: provision
+      schema:
+        description: Whether to provision the workflow as part of the request
+        type: boolean
+        default: false
+    flow_framework.create::query.validation:
+      in: query
+      name: validation
+      schema:
+        description: Whether to validate the workflow. Valid values are all (validate the template) and none (do not validate the template). Default is all.
+        type: string
+        default: all
+    flow_framework.update::query.update_fields:
+      in: query
+      name: update_fields
+      schema:
+        description: Whether to update only the fields included in the request body.
+        type: boolean
+        default: false
+
+    flow_framework.delete::query.clear_status:
+      in: query
+      name: clear_status
+      schema:
+        description: Determines whether to delete the workflow state (without deprovisioning resources) after deleting the template. OpenSearch deletes the workflow state only if the provisioning status is not IN_PROGRESS. Default is false.
+        type: boolean
+        default: false
+
+
+  requestBodies:
+    flow_framework.create:
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              name:
+                type: string
+                description: The name of the workflow.
```

Comment:
is there a spot for required/optional here?  I think name is required but none of the others are.

Reply:
```

            required:
              - name
```
Yes, you can check the required field below; it's addressed there